apps/omg_api/lib/state/transaction.ex

```diff
@@ -67,53 +62,51 @@ defmodule OMG.API.State.Transaction do
             %{
               blknum: pos_integer(),
               txindex: non_neg_integer(),
-              oindex: 0 | 1,
+              oindex: non_neg_integer(),
               currency: Crypto.address_t(),
               amount: pos_integer()
             }
           ],
-          [%{owner: Crypto.address_t(), amount: non_neg_integer()}],
-          non_neg_integer()
+          [%{owner: Crypto.address_t(), amount: non_neg_integer()}]
         ) :: {:ok, t()} | {:error, atom()}
-  def create_from_utxos(inputs, outputs, fee)
-  def create_from_utxos(inputs, _, _) when not is_list(inputs), do: {:error, :inputs_should_be_list}
-  def create_from_utxos(_, outputs, _) when not is_list(outputs), do: {:error, :outputs_should_be_list}
-  def create_from_utxos(inputs, _, _) when length(inputs) > @max_inputs, do: {:error, :too_many_inputs}
-  def create_from_utxos([], _, _), do: {:error, :at_least_one_input_required}
-  def create_from_utxos(_, outputs, _) when length(outputs) > @max_inputs, do: {:error, :too_many_outputs}
-  def create_from_utxos(_, _, fee) when fee < 0, do: {:error, :invalid_fee}
-
-  def create_from_utxos(inputs, outputs, fee) do
-    with {:ok, currency} <- validate_currency(inputs),
-         :ok <- validate_amount(inputs),
+  def create_from_utxos(inputs, outputs)
+  def create_from_utxos(inputs, _) when not is_list(inputs), do: {:error, :inputs_should_be_list}
+  def create_from_utxos(_, outputs) when not is_list(outputs), do: {:error, :outputs_should_be_list}
+  def create_from_utxos(inputs, _) when length(inputs) > @max_inputs, do: {:error, :too_many_inputs}
+  def create_from_utxos([], _), do: {:error, :at_least_one_input_required}
+  def create_from_utxos(_, outputs) when length(outputs) > @max_outputs, do: {:error, :too_many_outputs}
+
+  def create_from_utxos(input_utxos, outputs) do
+    with {:ok, currency} <- validate_currency(input_utxos, outputs),
+         :ok <- validate_amount(input_utxos),
          :ok <- validate_amount(outputs),
-         :ok <- amounts_add_up?(inputs, outputs, fee) do
-      {
-        :ok,
-        new(
-          inputs |> Enum.map(&{&1.blknum, &1.txindex, &1.oindex}),
-          currency,
-          outputs |> Enum.map(&{&1.owner, &1.amount})
-        )
-      }
+         :ok <- amounts_add_up?(input_utxos, outputs) do
+      {:ok,
+       new(
+         input_utxos |> Enum.map(&{&1.blknum, &1.txindex, &1.oindex}),
+         outputs |> Enum.map(&{&1.owner, currency, &1.amount})
+       )}
     end
   end
 
-  defp validate_currency(inputs) do
+  defp validate_currency(input_utxos, outputs) do
     currencies =
-      inputs
+      (input_utxos ++ outputs)
       |> Enum.map(& &1.currency)
-      |> Enum.uniq()
+      |> Enum.dedup()
 
-    if match?([_], currencies),
-      do: {:ok, currencies |> hd()},
-      else: {:error, :currency_mixing_not_possible}
+    # NOTE we support one currency
```

Comment:
do we still want this check?

Reply:
I kept it to not break convenience API.

Reply:
ah ok, because despite the flexible tx support, we still have single-currency transactions, gotcha